app/Controller/Component/RoleComponent.php

```diff
@@ -953,6 +1009,11 @@ public function isApproverForFlow($coPersonId, $coEfId, $coPetitionId=null) {
     }
     
     if($coEF['CoEnrollmentFlow']['approval_required']) {
+      // Does the user belong to the approvers group?
+      if($this->isCoOrCouApprover($coPersonId, $coEF['CoEnrollmentFlow']['co_id'])) {
+        $ret = true;
+      }
+
```

Comment:
This should go in the else clause below, with the following updated logic:

If no group is defined, then we use the following logic:
(1) $coPersonId is a CO admin
(2) authz_cou_id is specified and $coPersonId is a COU admin for that COU
(3) A COU is attached to the petition and $coPersonId is a COU admin
**(NEW) A COU is attached to the petition and $coPersonId is a member of the COU-specific approvers group**
(4) No authz_cou_id is specified and $coPersonId is a COU admin

Reply:
OR.